pkg/cloud/amazon/s3_storage.go

```diff
@@ -211,21 +214,26 @@ func S3URI(bucket, path string, conf *cloudpb.ExternalStorage_S3) string {
 }
 
 func parseS3URL(_ cloud.ExternalStorageURIContext, uri *url.URL) (cloudpb.ExternalStorage, error) {
+	s3URL := cloud.ConsumeURL{URL: uri}
```

Comment:
Why?

Reply:
So eventually I'm going to change our factory interface to not take `uri *url.URL` but a `cloud.ConsumeURL` instead but this will touch all storage providers. We use `ConsumeURL` so we can use the utility methods such as `ConsumeParam` over here - https://github.com/cockroachdb/cockroach/pull/85680/files#diff-45c45edcbc350e590cc007878194b26e66e945baed395f5812894335ff2e77ceR226. This will consume the param so that at the end of the parsing method we can validate that we have no invalid parameters that were passed in by the user. This is very similar to what cdc does in `changefeedccl/sink.go` and offers stronger validation.